src/overlays/actors/ovl_En_Osn/z_en_osn.c

```diff
@@ -74,59 +101,841 @@ static DamageTable D_80AD2550 = {
+                case 7:
+                    this->unk_1EC = 4;
+                    break;
+                case 8:
+                    this->unk_1EC = 0;
+                    break;
+                case 10:
+                    this->unk_1EC = 8;
+                    break;
+                case 11:
+                    this->unk_1EC = 9;
+                    break;
+                case 13:
+                    this->unk_1EC = 0xA;
+                    break;
+                case 15:
+                    this->unk_1EC = 0xB;
+                    break;
+                case 16:
+                    this->unk_1EC = 0xC;
+                    break;
+                case 17:
+                    this->unk_1EC = 0xD;
+                    break;
+                case 18:
+                    this->unk_1EC = 0xE;
+                    break;
+                case 19:
+                    this->unk_1EC = 0x11;
+                    break;
+                case 20:
+                    this->unk_1EC = 0x12;
+                    break;
+                case 21:
+                    this->unk_1EC = 0x14;
+                    break;
+                case 22:
+                    this->unk_1EC = 0x15;
+                    break;
+                case 23:
+                    this->unk_1EC = 0x17;
+                    break;
+                case 24:
+                    this->unk_1EC = 0x18;
+                    break;
+                default:
+                    this->unk_1EC = 0;
+                    break;
+            }
+            Actor_ChangeAnimation(&this->skelAnime, sAnimations, this->unk_1EC);
+        }
+
+        if ((this->unk_1EC == 5) && (globalCtx->sceneNum == 8) && (gSaveContext.sceneSetupIndex == 0xB) &&
+            (globalCtx->csCtx.frames == 0x190)) {
+            Actor_PlaySfxAtPos(&this->actor, 0x697D);
+        }
+        if (this->unk_1EC == 0x12) {
+            func_80AD0998(this);
+        }
+        if (this->unk_1EC == 0x15) {
+            func_80AD0A24(this);
+        }
+        if (this->unk_1EC == 0x18) {
+            func_80AD0AB0(this);
+        }
+        if ((this->unk_1EC == 0x14) &&
+            (((Animation_OnFrame(&this->skelAnime, 17.0f))) || (Animation_OnFrame(&this->skelAnime, 27.0f)) ||
+             (Animation_OnFrame(&this->skelAnime, 37.0f)) || (Animation_OnFrame(&this->skelAnime, 47.0f)) ||
+             (Animation_OnFrame(&this->skelAnime, 57.0f)) || (Animation_OnFrame(&this->skelAnime, 67.0f)))) {
+            Actor_PlaySfxAtPos(&this->actor, 0x29B3);
+        }
+        func_800EDF24(&this->actor, globalCtx, temp_v0);
+        return;
+    }
+    this->unk_1F0 = 1;
+    this->unk_1ED = 0x63;
+    func_80AD144C(this, globalCtx);
+}
+
+void func_80AD19A0(EnOsn* this, GlobalContext* globalCtx) {
+    u8 temp_v0 = Message_GetState(&globalCtx->msgCtx);
+
+    if ((temp_v0 == 6 || temp_v0 == 5) && func_80147624(globalCtx)) {
+        if (this->unk_1EA & 0x20) {
+            this->unk_1EA &= 0xFFDF;
```

Comment:
```suggestion
            this->unk_1EA &= ~0x20;
```
Might have to add a (s16) cast for this one to match